api/server/server.go

```diff
@@ -1640,8 +1640,9 @@ func allocateDaemonPort(addr string) error {
 		return fmt.Errorf("failed to lookup %s address in host specification", host)
 	}
 
+	pa := portallocator.Get()
```

Comment:
Thanks @stevvooe. The port allocator package needs a redesign to reflect few other issues as captured in (https://github.com/docker/libnetwork/issues/169). This PR tries to port the existing functionality without impacting the behavior so that we have apples-apples comparison. If you are okay, we can address your concern along with the above enhancement. Please let us know.